### PR TITLE
add connect timeout, support accurate send timeout

### PR DESCRIPTION
**Pull request by lihanharry**

### CODE REVIEW

**Comment by jeking3** [comment]
THRIFT-4171

**Comment by jeking3** [comment]
Please squash your changes to a single commit if possible.  Thanks.

**Comment by dcelasun** [comment]
Please don't change the code style while you are doing unrelated changes, it makes it very hard to review what *actually* changed. Even with the changes, there are lots of style inconsistencies like opening braces for functions are sometimes on the same line, sometimes on a new line.

So please revert any style-only changes and follow the existing style. If you want to also change/improve code style, open a separate PR for that. Thank you.

**Comment by jeking3** [comment]
Fortunately these changes are confined to one file so reviewing it isn't too bad - hopefully they pass the new php formatting tests in the sca build.

**Comment by jeking3** [comment]
The new style changes and checks are pretty recent, based on work done by @RobberPhex.

File: lib/php/lib/Transport/TSocket.php [hunk 1]
```diff
@@ -54,6 +54,24 @@ class TSocket extends TTransport
...
+    protected $connTimeoutSec_ = 0;
+
+    /**
+     * Connect timeout in microseconds.
+     *
+     * Combined with connTimeoutUsec this is used for connect timeouts.
+     *
+     * @var int
+     */
+    protected $connTimeoutUsec_ = 100000;
```
Comment (jeking3) [inline lib/php/lib/Transport/TSocket.php]: So the default timeout if not otherwise specified is 100ms?  Sounds bad.

**Comment by jeking3** [comment]
Reminder this still needs some work.

**Comment by jeking3** [comment]
@lihanharry Reminder this still needs some work.  Will you be able to complete this?

**Comment by jeking3** [comment]
Trying one more time... @lihanharry Reminder this still needs some work. Will you be able to complete this?

**Pull request closed without merging**